public/js/search.js

```diff
@@ -0,0 +1,66 @@
+$(function(config){
+  var algolia = algoliasearch(config.applicationId, config.apiKey);
+  var index = algolia.initIndex(config.indexName);
+
+  hideSearch();
+
+  $('.clear-search img, .search-results-container img').on('click', function() {
+    $('.sidebar-navigation--search input').val('');
+    hideSearch();
+  });
+
+  $('.sidebar-navigation--search input').on('input', function() {
```

Comment:
Yeah, this event is fired only when an input element's value attribute has changed:
https://developer.mozilla.org/en-US/docs/Web/Events/input